### ISSUE TITLE
Disable theming by default in Linux distributions?

### ISSUE THREAD

**Issue by mickaelistria** [eclipse-platform/eclipse.platform.ui#629]
Currently, on Linux, most contributors seem to either use dark theme or use no theme. There seems to be a relative consensus that the light theme on Linux is not better than no theme. So I would recommend that -on Linux only- we just disable theming by default.
Thoughts?

**Comment by trancexpress** [comment]
+1

**Comment by AObuchow** [comment]
I think this would solve the bad first impression some users get when using a non-light GTK theme alongside the default Eclipse light theme (gives a bad mixture of light and dark colours used that is unusable) so +1.

As long as users are able to still switch to the light theme if desired, this seems good to me.

**Comment by laeubi** [comment]
> Currently, on Linux, most contributors seem to either use dark theme or use no theme. There seems to be a relative consensus that the light theme on Linux is not better than no theme.

Maybe it helps to post a screenshot of how they currently look like. How do I know what is used? I usually download Eclipse and use it "as-is" ...

**Comment by iloveeclipse** [comment]
@laeubi : they are looking mostly same if not using "dark" or some other crazy system theme.

With theming: (clearlooks)
![image](https://user-images.githubusercontent.com/964108/224751790-090bb5b6-4cc1-47c6-b4d4-a11047d3c495.png)

Without:
![image](https://user-images.githubusercontent.com/964108/224752008-3b516124-3531-48c8-b61d-6c132188399d.png)

@mickaelistria : I see only benefits by disabling themes, but that is independently on the OS.

Om Linux we saw extreme slowdown with theming enabled in the past (may be also today) if running Eclipse over ssh, and switched globally themes off because of that. I personally never regret that. I also see some users prefer to use dark theme, but here again, it will not change much for them, as they will have to enable themes AND select Dark theme.

**Comment by mickaelistria** [comment]
>  I see only benefits by disabling themes, but that is independently on the OS.

I wouldn't mind disabling theming by default for all OS; but I'm not much familiar with how Eclipse IDE looks on other OS with/without theme to promote the idea there.

**Comment by Phillipus** [comment]
Some people still love the Classic theme... 😲

**Comment by Phillipus** [comment]
Will this proposed change affect RCP apps? Our RCP app defaults to light theme on all OS. It might come as a surprise to our Linux users...

**Comment by mickaelistria** [comment]
> Will this proposed change affect RCP apps? Our RCP app defaults to light theme on all OS

Yes, RCP apps would have to explicitly enabled theming if they want it on Linux.

> It might come as a surprise to our Linux users...

Surprise can be good.

**Comment by Phillipus** [comment]
> Yes, RCP apps would have to explicitly enabled theming if they want it on Linux.

In that case perhaps do it the other way around and explicitly disable theming for Eclipse distributions? Changes that might be OK for Eclipse IDE might not be appropriate for RCP apps.

**Comment by mickaelistria** [comment]
> In that case perhaps do it the other way around and explicitly disable theming for Eclipse distributions? Changes that might be OK for Eclipse IDE might not be appropriate for RCP apps.

The Platform is meant to lead; the UI layout is not an API contract. If there is a consensus among Platform maintainer that 1 layout is better, the Platform can change.
RCP apps that want everything set in stone on their end need should make the effort on their end instead of assuming Platform will never change.

**Comment by merks** [comment]
I guess this can't be done via /org.eclipse.sdk/plugin_customization.ini like one could normally do because you can't do something conditionally there...

**Comment by mickaelistria** [comment]
> I guess this can't be done via /org.eclipse.sdk/plugin_customization.ini like one could normally do because you can't do something conditionally there...

Indeed, that's the issue.
